kratos/geometries/quadrilateral_3d_4.h

```diff
@@ -649,16 +640,17 @@ template<class TPointType> class Quadrilateral3D4
      * @see DeterminantOfJacobian
      * @see InverseOfJacobian
      */
-    JacobiansType& Jacobian( JacobiansType& rResult,
-                                     IntegrationMethod ThisMethod ) const override
+    JacobiansType& Jacobian( 
+        JacobiansType& rResult,
+        IntegrationMethod ThisMethod 
+        ) const override
     {
-        //getting derivatives of shape functions
+        // Getting derivatives of shape functions
         ShapeFunctionsGradientsType shape_functions_gradients =
-            CalculateShapeFunctionsIntegrationPointsLocalGradients( ThisMethod );
-        //getting values of shape functions
+        CalculateShapeFunctionsIntegrationPointsLocalGradients( ThisMethod );

```

Comment:
You should not use the private calculate method. You have the calculated values in geometry data and can be used via ShapeFunctionsIntegrationPointsLocalGradients which return a reference. So no copy is needed!!

Reply:
Ok

Reply:
Should I change this in all geometries?